### DISCUSSION TITLE
How to Create a System Login with Role in Laravel?

### DISCUSSION THREAD

**Discussion by koj4** [community/community#147415] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am developing a Laravel application that requires a login feature with several roles, such as Admin and User. I want each role to have access to a specific page. How to implement this system in the best way?

**Comment by hkqr** [comment]


**Reply by koj4** [reply]
Thank you very much, really great help!

**Comment by Abdelmonem-Dev** [comment]
I'm very late but no problem, The best option I can recommend to you is through permission, you can store user permissions as an integer using bitwise operations. This method allows you to efficiently store multiple permissions in a single column.

A user with View Posts (1) and Edit Posts (2) would have a stored value of 3 (0001 + 0010 = 0011).
A user with Delete Posts (4) and Manage Users (8) would have a stored value of 12 (0100 + 1000 = 1100).

You can read more about this topic to get a clearer picture.

and in Laravel you have a library for permissions.
